pandas/tests/reshape/test_pivot.py

```diff
@@ -2378,6 +2378,47 @@ def test_pivot_table_with_mixed_nested_tuples(self, using_array_manager):
             expected["small"] = expected["small"].astype("int64")
         tm.assert_frame_equal(result, expected)
 
+    def test_pivot_table_aggfunc_nunique_with_different_values(self):
+        np.random.seed(0)
+        test = DataFrame(
+            {
+                "a": np.random.randint(0, 10, size=(10,)),
+                "b": np.random.randint(0, 10, size=(10,)),
+                "c": np.random.randint(0, 10, size=(10,)),
+                "d": np.random.randint(0, 10, size=(10,)),
+            }
+        )
+        expected = test.pivot_table(
```

Comment:
Can you construct this without `pivot_table` i.e. `expected = pd.DataFrame(..., columns=...)`